Add phrase search (#1021)
meilisearch/documentation: reference/api/search.md

[Learn more about how to use the search parameters](/reference/features/search_parameters.md).

#### Placeholder search

> When the `q` parameter is not specified, a [placeholder](/reference/features/search_parameters.md#query-q) search is run instead.

Reviewer: ```suggestion
When no search query is specified, a [placeholder search](/reference/features/search_parameters.md#query-q) is run instead.
```

Reviewer: Not relevant to this PR, but I want to make this change since `q` parameter isn't really exposed to users of the SDKs (they have dedicated search functions).

#### Phrase search

> Query terms enclosed in double quotes are parsed as [phrase searches](/reference/features/search_parameters.md#query-q).

Reviewer: ```suggestion
Query terms enclosed in double quotes are treated as [phrase searches](/reference/features/search_parameters.md#query-q).
```

When the `q` parameter is not specified, a [placeholder](/reference/features/search_parameters.md#query-q) search is run instead.

#### Phrase search

Query terms enclosed in double quotes are parsed as [phrase searches](/reference/features/search_parameters.md#query-q).

> 

Author: What do you think about having this content repeated on both POST and GET? I'm not usually a fan of duplicated content, but having it on only one might give the impression that the two routes do not offer the same functionality.

> 

Reviewer: Yes sounds good, we're just going to delete GET eventually anyway so no harm in repeating it.

> When the `q` parameter is not specified, a [placeholder](/reference/features/search_parameters.md#query-q) search is run instead.

Reviewer: ```suggestion
When no search query is specified, a [placeholder search](/reference/features/search_parameters.md#query-q) is run instead.
```

#### Phrase search

> Query terms enclosed in double quotes are parsed as [phrase searches](/reference/features/search_parameters.md#query-q).

Reviewer: ```suggestion
Query terms enclosed in double quotes are treated as [phrase searches](/reference/features/search_parameters.md#query-q).
```
